core/lib/src/config/config.rs

```diff
@@ -39,9 +40,9 @@ pub struct Config {
     /// The environment that this configuration corresponds to.
     pub environment: Environment,
     /// The address to serve on.
-    pub address: String,
+    pub address: Address,
```

Comment:
Because this is `pub`, a user can do something like:

```rust
let mut config = Config::new(..);
config.address = Address::from_str("unix:foo");
rocket::custom(config).launch();
```

On Windows, this will reach the `unreachable!()` in the `serve_unix_socket` macro.

To prevent this, (instead removing the `pub`) we should should ensure that `Address` doesn't parse `unix:` on non-Unix. We should also remove the `Address::Unix` variant entirely on non-Unix.